spec.html

```diff
@@ -35827,16 +35656,27 @@ <h1>Properties of Map Instances</h1>
       <h1>Map Iterator Objects</h1>
       <p>A Map Iterator is an object, that represents a specific iteration over some specific Map instance object. There is not a named constructor for Map Iterator objects. Instead, map iterator objects are created by calling certain methods of Map instance objects.</p>
 
-      <emu-clause id="sec-createmapiterator" aoid="CreateMapIterator">
+      <emu-clause id="sec-createmapiterator" aoid="CreateMapIterator" oldids="sec-properties-of-map-iterator-instances,table-50,table-internal-slots-of-map-iterator-instances">
         <h1>CreateMapIterator ( _map_, _kind_ )</h1>
         <p>The abstract operation CreateMapIterator takes arguments _map_ and _kind_. This operation is used to create iterator objects for Map methods that return such iterators. It performs the following steps when called:</p>
         <emu-alg>
           1. Perform ? RequireInternalSlot(_map_, [[MapData]]).
-          1. Let _iterator_ be ! OrdinaryObjectCreate(%MapIteratorPrototype%, &laquo; [[IteratedMap]], [[MapNextIndex]], [[MapIterationKind]] &raquo;).
-          1. Set _iterator_.[[IteratedMap]] to _map_.
-          1. Set _iterator_.[[MapNextIndex]] to 0.
-          1. Set _iterator_.[[MapIterationKind]] to _kind_.
-          1. Return _iterator_.
+          1. Let _closure_ be a new Abstract Closure with no parameters that captures _map_ and _kind_ and performs the following steps when called:
+            1. Let _entries_ be the List that is _map_.[[MapData]].
+            1. Let _index_ be *+0*.
```

Comment:
```suggestion
            1. Let _index_ be 0.
```

You're using this to index a List, and Lists are indexed by mathematical values, not Numbers.